src/PowerShellEditorServices/Extensions/Api/EditorUIService.cs

```diff
@@ -0,0 +1,171 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Services.PowerShellContext;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PowerShellEditorServices.Extensions.Api
+{
+    /// <summary>
+    /// Object specifying a UI prompt option to display to the user.
+    /// </summary>
+    public sealed class PromptChoiceDetails
+    {
+        /// <summary>
+        /// Construct a prompt choice object for display in a prompt to the user.
+        /// </summary>
+        /// <param name="label">The label to identify this prompt choice. May not contain commas (',').</param>
+        /// <param name="helpMessage">The message to display to users.</param>
+        public PromptChoiceDetails(string label, string helpMessage)
+        {
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            if (label.Contains(","))
+            {
+                throw new ArgumentException($"Labels may not contain ','. Label: '{label}'", nameof(label));
+            }
+
+            Label = label;
+            HelpMessage = helpMessage;
+        }
+
+        /// <summary>
+        /// The label to identify this prompt message.
+        /// </summary>
+        public string Label { get; }
+
+        /// <summary>
+        /// The message to display to users in the UI for this prompt choice.
+        /// </summary>
+        public string HelpMessage { get; }
+    }
+
+    /// <summary>
+    /// A service to manipulate the editor user interface.
+    /// </summary>
+    public interface IEditorUIService
```

Comment:
This lets you send prompt requests

Reply:
Also need `powerShell/showInputPrompt` (used in "extract to function" to name the function for example)

Reply:
Found it